### PR TITLE
feat/open-in-explorer-for-wallets-on-account-screen

### PR DESCRIPTION
**Pull request by NikIvv**

### CODE REVIEW

**Comment by NikIvv** [comment]
Some notes on the current implementations: explorer.btc.com was not accessible during my testing; removed 'Open In Explorer' menu option for USDT and USDC as they have not been described in configs.

File: src/components/ShareURIDialog.vue [hunk 1]
```diff
@@ -86,6 +97,14 @@ export default {
...
       this.showQrcodeRendererDialog = true
       this.show = false
+    },
+    openInExplorer() {
+      const explorerUri = config[this.crypto.toLowerCase()].explorerAddress.replace(
+        /\${([a-zA-Z_]+?)}/,
+        this.address
+      )
+      const explorerLink = this.isADM ? websiteUriToOnion(explorerUri) : explorerUri
+      window.open(explorerLink, '_blank', 'resizable,scrollbars,status,noopener')
```
Comment (bludnic) [inline src/components/ShareURIDialog.vue:107]: You can use [getExplorerAddressUrl](https://github.com/Adamant-im/adamant-im/blob/eacc2d3fd8c6a3a14dd532f5d560783f4d56a967/src/config/utils/getExplorerAddressUrl.ts) helper:

1. No need in `websiteUriToOnion`. Tor has a separate [config](https://github.com/Adamant-im/adamant-im/blob/658efd2b068473745615f6dfe4e4b097e62c4c27/src/config/tor.json#L51) which is automatically replaced when building the app.
2. Use [isErc20](https://github.com/Adamant-im/adamant-im/blob/e0a8fdb2980e6e85013726826f9d33dea3f0a668/src/lib/constants/index.js#L80) helper to detect ETH based crypto.

File: src/components/ShareURIDialog.vue [hunk 2]
```diff
@@ -22,6 +22,14 @@
               {{ $t('home.show_qr_code') }}
             </v-list-item-title>
           </v-list-item>
+          <v-list-item
+            v-if="crypto !== cryptos.USDT && crypto !== cryptos.USDC"
```
Comment (bludnic) [inline src/components/ShareURIDialog.vue:26]: You don't need this. See comment below (`isErc20` helper)

File: src/components/ShareURIDialog.vue [hunk 3]
```diff
@@ -57,7 +67,8 @@ export default {
   emits: ['update:modelValue'],
   data: () => ({
     className: 'share-uri-dialog',
-    showQrcodeRendererDialog: false
+    showQrcodeRendererDialog: false,
+    cryptos: Cryptos
```
Comment (bludnic) [inline src/components/ShareURIDialog.vue:71]: You probably don't need this anymore

File: src/components/ShareURIDialog.vue [hunk 4]
```diff
@@ -32,7 +40,9 @@
 <script>
```
Comment (bludnic) [inline src/components/ShareURIDialog.vue:40]: After fixing the issues, I suggest you to do a little refactor by rewriting the component using Composition API and TypeScript. Follow the guidelines https://developers.adamant.im/template-classes.html

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/components/ShareURIDialog.vue (hunk 1 above)
Comment (bludnic) [reply src/components/ShareURIDialog.vue:107]: We build the app for Tor using [tor.json](https://github.com/Adamant-im/adamant-im/blob/eacc2d3fd8c6a3a14dd532f5d560783f4d56a967/src/config/tor.json#L30) config, so the `getExplorerAddressUrl()` will choose the right URL automatically.

**Comment by bludnic** [comment]
> Some notes on the current implementations: explorer.btc.com was not accessible during my testing; removed 'Open In Explorer' menu option for USDT and USDC as they have not been described in configs.

@NikIvv Bitcoin explorer was replaced in https://github.com/Adamant-im/adamant-wallets/pull/101
We will include these changes in the next release.

File: src/components/ShareURIDialog.vue [hunk 5]
```diff
@@ -22,6 +22,11 @@
               {{ $t('home.show_qr_code') }}
             </v-list-item-title>
           </v-list-item>
+          <v-list-item v-if="!isErc" @click="openInExplorer">
```
Comment (bludnic) [inline src/components/ShareURIDialog.vue:25]: ERC20 tokens also have an address (same as for ETH).
Handle that logic in the `openInExplorer` method.

**Review by bludnic — APPROVED** [review]

**Pull request merged**